src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -141,12 +147,134 @@ class PortIteratorHelper : public PortMapHelper {
     bool sliced_src;
     dnnl::memory full_mem;
 
+    dnnl::reorder reorder;
+    dnnl::memory mem_holder_src;
+    dnnl::memory mem_holder_dst;
+
     int iter_count;
 };
 
+class InputEdgePortHelper : public PortMapHelper {
+public:
+    InputEdgePortHelper(const MemoryPtr &from, const MemoryPtr &to, const dnnl::engine& eng) {
+        mem_holder_src = from->GetPrimitive();
+        mem_holder_dst = to->GetPrimitive();
+        //if (mem_holder_src.get_desc() != mem_holder_dst.get_desc()) {
+            reorder = {mem_holder_src, mem_holder_dst};
+        //} else {
+        //    mem_holder_dst.set_data_handle(mem_holder_src.get_data_handle());
+        //}
+    }
+
+    void execute(dnnl::stream strm, int iter = -1) override {
+        if (iter != 0 && reorder.get(true)) {
+            reorder.execute(strm, mem_holder_src, mem_holder_dst);
+        }
+    }
+
+private:
+    dnnl::reorder reorder;
+    dnnl::memory mem_holder_src;
+    dnnl::memory mem_holder_dst;
+};
+
 class BackEdgePortHelper : public PortMapHelper {
 public:
-    BackEdgePortHelper(const MemoryPtr &from, const MemoryPtr &to, const dnnl::engine& eng) {
+    BackEdgePortHelper(const Node* from, Node* to) : m_from(from), m_to(to) {
+        auto from_desc = m_from->getBaseMemDescAtInputPort(inputNodePortIdx);
+        auto to_desc = m_to->getBaseMemDescAtOutputPort(outputNodePortIdx);
+        VectorDims commonDims;
+        if (from_desc->isDefined()) {
+            commonDims = from_desc->getShape().getStaticDims();
+        } else if (to_desc->isDefined()) {
+            commonDims = to_desc->getShape().getStaticDims();
+        } else {
+            auto& dimsFromMax = from_desc->getShape().getMaxDims();
+            auto& dimsToMax = to_desc->getShape().getMaxDims();
+            VectorDims commonMax(dimsFromMax.size());
+            for (size_t i = 0; i < dimsFromMax.size(); ++i) {
+                commonMax[i] = std::min(dimsFromMax[i], dimsToMax[i]);
+            }
+
+            auto& dimsFromMin = from_desc->getShape().getMinDims();
+            auto& dimsToMin = to_desc->getShape().getMinDims();
+            VectorDims commonMin(dimsFromMin.size());
+            for (size_t i = 0; i < dimsFromMax.size(); ++i) {
+                commonMin[i] = std::max(dimsFromMin[i], dimsToMin[i]);
+            }
+
+            Shape commonShape(commonMin, commonMax);
+            commonDims = MemoryDescUtils::makeDummyShape(commonShape).getStaticDims();
+        }
+
+        if (!from_desc->isDefined()) {
```

Comment:
BackEdgePortHelper is constructed in CreatePrimitives() where in dynamic shape cases they might not be defined.
Cannot understand why do you have this commonDims code snippet.
.